pr_agent/algo/utils.py

```diff
@@ -661,3 +662,12 @@ def find_line_number_of_relevant_line_in_file(diff_files: List[FilePatchInfo],
                             absolute_position = start2 + delta - 1
                             break
     return position, absolute_position
+
+def github_output(output_data: dict, key_name: str):
+    if get_settings().github_action_config.enable_output is False:
+        return
+    
+    output = os.getenv('GITHUB_OUTPUT')
+    key_data = output_data.get(key_name, {})
+    with open(output, 'w') as f:
+        f.write(f"{key_name}={json.dumps(key_data, indent=None, ensure_ascii=False)}")
```

Comment:
(1)
change name to 'github_action_output'
it can be confusing to call it as a generic 'github_output' name

(2) wrap everything with try-except.
see also PR-Agent feedback:
"
Environment Variable Dependency: The feature relies on the GITHUB_OUTPUT environment variable being set.
"
Will it always be set ?

(3)
change
"    if get_settings().github_action_config.enable_output is False:
        return"
        
 to getter only:
if get_settings().get("github_action_config.enable_output", False)

Reply:
@mrT23 Thank you for your reivewing.

- (1) : fixed.
- (2) : fixed, refined & add test code. 
I think `GITHUB_OUTPUT` env is always available on github actions runner. Refined the output implementation based on most using codebases. ([ref.1](https://github.com/orgs/community/discussions/28146#discussioncomment-4110404) / [ref.2](https://github.com/search?q=os.environ%5B%27GITHUB_OUTPUT%27%5D+language%3AYAML&type=code&p=3&l=YAML))
Note that this feature is implicitly disabled in the configuration if use in other than github actions. so, I think no side-effect.
- (3): fixed.